pandas/core/groupby/generic.py

```diff
@@ -1813,9 +1813,20 @@ def groupby_series(obj, col=None):
             # Try to consolidate with normal wrapping functions
             from pandas.core.reshape.concat import concat
 
-            results = [groupby_series(content, label) for label, content in obj.items()]
+            axis_number = obj._get_axis_number(self.axis)
```

Comment:
@WillAyd this is adding quite a bit of very hard and bespoke code

pls don’t merge non trivial things like this

this is also likely non performant

pls revert